src/titiler/xarray/titiler/xarray/io.py

```diff
@@ -0,0 +1,273 @@
+    consolidated: Optional[bool] = True,
+    cache_client: Optional[CacheClient] = None,
+) -> xarray.Dataset:
+    """Open dataset."""
+    # Generate cache key and attempt to fetch the dataset from cache
+    if cache_client:
+        cache_key = f"{src_path}_{group}" if group is not None else src_path
+        data_bytes = cache_client.get(cache_key)
+        if data_bytes:
+            return pickle.loads(data_bytes)
+
+    protocol = parse_protocol(src_path, reference=reference)
+    xr_engine = xarray_engine(src_path)
+    file_handler = get_filesystem(src_path, protocol, xr_engine)
+
+    # Arguments for xarray.open_dataset
+    # Default args
+    xr_open_args: Dict[str, Any] = {
+        "decode_coords": "all",
+        "decode_times": decode_times,
+    }
+
+    # Argument if we're opening a datatree
+    if isinstance(group, int):
+        xr_open_args["group"] = group
+
+    # NetCDF arguments
+    if xr_engine == "h5netcdf":
+        xr_open_args["engine"] = "h5netcdf"
+        xr_open_args["lock"] = False
+    else:
+        # Zarr arguments
+        xr_open_args["engine"] = "zarr"
+        xr_open_args["consolidated"] = consolidated
+
+    # Additional arguments when dealing with a reference file.
+    if reference:
+        xr_open_args["consolidated"] = False
+        xr_open_args["backend_kwargs"] = {"consolidated": False}
+
+    ds = xarray.open_dataset(file_handler, **xr_open_args)
+
+    if cache_client:
+        # Serialize the dataset to bytes using pickle
+        data_bytes = pickle.dumps(ds)
+        cache_client.set(cache_key, data_bytes)
+
+    return ds
+
+
+def arrange_coordinates(da: xarray.DataArray) -> xarray.DataArray:
+    """Arrange coordinates to DataArray.
+
+    An rioxarray.exceptions.InvalidDimensionOrder error is raised if the coordinates are not in the correct order time, y, and x.
+    See: https://github.com/corteva/rioxarray/discussions/674
+
+    We conform to using x and y as the spatial dimension names. You can do this a bit more elegantly with metpy but that is a heavy dependency.
+
+    """
+    if "x" not in da.dims and "y" not in da.dims:
+        latitude_var_name = "latitude" if "latitude" in da.dims else "lat"
+        longitude_var_name = "longitude" if "longitude" in da.dims else "lon"
+        da = da.rename({latitude_var_name: "y", longitude_var_name: "x"})
+
+    if "time" in da.dims:
+        da = da.transpose("time", "y", "x")
+    else:
+        da = da.transpose("y", "x")
+
+    return da
+
+
+def get_variable(
+    ds: xarray.Dataset,
+    variable: str,
+    datetime: Optional[str] = None,
+    drop_dim: Optional[str] = None,
+) -> xarray.DataArray:
+    """Get Xarray variable as DataArray."""
+    da = ds[variable]
+    da = arrange_coordinates(da)  # TODO: Duplicated call L173
+
+    # TODO: add test
+    if drop_dim:
+        dim_to_drop, dim_val = drop_dim.split("=")
+        da = da.sel({dim_to_drop: dim_val}).drop(dim_to_drop)
+
+    da = arrange_coordinates(da)
```

Comment:
from titiler-xarray repo: we have twice a call to arrange_coordinates... not sure why cc @abarciauskas-bgse

Reply:
🤔 seems like an error, we can probably safely remove it in line 170